packages/back-end/src/models/EventWebhookModel.ts

```diff
@@ -393,8 +393,8 @@ export const getAllEventWebHooksForEvent = async ({
   });
 
   const docs = allDocs.filter((doc) => {
-    if (!filterOptional(tags, doc.tags)) return false;
-    if (!filterOptional(projects, doc.projects)) return false;
+    if (!filterOptional(doc.tags, tags)) return false;
+    if (!filterOptional(doc.projects, projects)) return false;
```

Comment:
The logic was inverted here: we want webhooks with no projects to match all events regardless of their projects and same for tags.